core/shared/src/main/scala/org/http4s/headers/Keep-Alive.scala

```diff
@@ -52,11 +52,11 @@ keep-alive-extension = token [ "=" ( token / quoted-string ) ]
       extension: List[(String, Option[String])],
   ): ParseResult[`Keep-Alive`] =
     if (timeoutSeconds.isDefined || max.isDefined || extension.nonEmpty) {
-      val reservedTokens = List("token", "max")
-      if (extension.exists(p => reservedTokens.contains(p._1))) {
+      val reservedToken = "token"
```

Comment:
If the intention of this PR is to drop support for the deprecated `max` then IMO it should be dropped completely (including the `case class Max` the `max` parameter in `apply` etc), not just from the extension list. But that breaks bin-compat. So maybe it would be better to not remove the max at all? If we remove it only from the list of extensions then we end up with the strange situation where max can be define in two different places.

Reply:
Good shout. I wasn't aware of the places where it exists. Thoughts @armanbilge ?

Reply:
Right, we can fix this on 1.x however.

The part I still haven't grokked is how this change relates to the flaky test?